mmv1/products/clouddeploy/api.yaml

```diff
@@ -0,0 +1,218 @@
+      - !ruby/object:Api::Type::String
+        name: 'etag'
+        output: true
+        description: |
+          This checksum is computed by the server based on the value of other fields, and may be sent on update and delete requests to ensure the client has an up-to-date value before proceeding.
+      - !ruby/object:Api::Type::KeyValuePairs
+        name: 'labels'
+        description: |
+          Resource labels to represent user-provided metadata.
+      - !ruby/object:Api::Type::KeyValuePairs
+        name: 'annotations'
+        description: |
+          User annotations. These attributes can only be set and used by the user, and not by Google Cloud Deploy.
+      - !ruby/object:Api::Type::Boolean
+        name: 'requireApproval'
+        description: |
+          Whether or not the Target requires approval.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'gke'
+        description: |
+          Information specifying a GKE Cluster.
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'cluster'
+            exactly_one_of:
+              - gke.0.cluster
+              - anthos_cluster.0.membership
+            description: |
+              Information specifying a GKE Cluster. Format is `projects/{projectId}/locations/{locationId}/clusters/{cluster_id}.
+          - !ruby/object:Api::Type::Boolean
+            name: 'internalIp'
+            description: |
+              If true, cluster is accessed using the private IP address of the control plane endpoint. Otherwise, the default IP address of the control plane endpoint is used.
+              The default IP address is the private IP address for clusters with private control-plane endpoints and the public IP address otherwise.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'anthosCluster'
+        description: |
+          Information specifying a GKE Cluster.
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'membership'
+            exactly_one_of:
+              - gke.0.cluster
+              - anthos_cluster.0.membership
+            description: |
+              Membership of the GKE Hub-registered cluster to which to apply the Skaffold configuration. Format is projects/{project}/locations/{location}/memberships/{membership_name}.
+      - !ruby/object:Api::Type::Array
+        name: 'executionConfigs'
+        description: |
+          Configurations for all execution that relates to this Target. Each ExecutionEnvironmentUsage value may only be used in a single configuration;
+          using the same value multiple times is an error. When one or more configurations are specified, they must include the RENDER and DEPLOY ExecutionEnvironmentUsage values.
+          When no configurations are specified, execution will use the default specified in DefaultPool.
```

Comment:
Can you add a test that doesn't specify an executionConfig? I'm curious what will happen when none is specified, will the API add a default one to this value or just handle the behavior without modifying this field

Reply:
Done.